File: beakjoon/11723.py
# 파이썬이라 그런지 시간초과로 안됨.

n = int(input())
isIn = [False for _ in range(21)]

for _ in range(n):
    ipt = list(input().split())
    if len(ipt) == 1:
        cmd = ipt[0]
    else:
        cmd, nums = ipt
        nums = int(nums)
        c = [0 for _ in range(nums)]

    if cmd == 'all':
        for i in range(1, 21):
            isIn[i] = True
    elif cmd == 'empty':
        for i in range(1, 21):
            isIn[i] = False
    elif cmd == 'add':
        isIn[nums] = True
    elif cmd == 'remove':
        isIn[nums] = False
    elif cmd == 'check':
        if (isIn[nums]):
            print(1)
        else:
            print(0)
    else:
        isIn[nums] = 1 - isIn[nums]

File: beakjoon/11723.cpp
#include <cstdio>
#include <cstring>
using namespace std;

const char* ADD = "add";
const char* REMOVE = "remove";
const char* CHECK = "check";
const char* TOGGLE = "toggle";
const char* ALL = "all";
const char* EMPTY = "empty";

char cmm[10];
int n,num, sets[21];

int main()
{
    for(int i=1; i<=20; ++i) sets[i] = 0;
    scanf("%d",&n);
    
    for(int ct=0; ct<n; ++ct) {
        scanf("%s",cmm);
        if(!(cmm == ALL || cmm == EMPTY)) scanf("%d",&num);
        if(!strcmp(cmm,ADD)) sets[num] = 1;
        else if(!strcmp(cmm,REMOVE)) sets[num] = 0;
        else if(!strcmp(cmm,CHECK)) sets[num] ? printf("1\n") : printf("0\n");
        else if(!strcmp(cmm,TOGGLE)) sets[num] = 1-sets[num];
        else if(!strcmp(cmm,ALL))
            for(int i=1; i<=20; ++i) sets[i] = 1;
        else
            for(int i=1; i<=20; ++i) sets[i] = 0;
    }   
}